go/tasks/plugins/k8s/kfoperators/common/common_operator.go

```diff
@@ -0,0 +1,125 @@
+package common
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	commonOp "github.com/kubeflow/tf-operator/pkg/apis/common/v1"
+	logUtils "github.com/lyft/flyteidl/clients/go/coreutils/logs"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	flyteerr "github.com/lyft/flyteplugins/go/tasks/errors"
+	"github.com/lyft/flyteplugins/go/tasks/logs"
+	pluginsCore "github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	TensorflowTaskType = "tensorflow"
+	PytorchTaskType    = "pytorch"
+)
+
+func ExtractCurrentCondition(jobConditions []commonOp.JobCondition) (commonOp.JobCondition, error) {
+	sort.Slice(jobConditions[:], func(i, j int) bool {
+		return jobConditions[i].LastTransitionTime.Time.After(jobConditions[j].LastTransitionTime.Time)
+	})
+
+	for _, jc := range jobConditions {
+		if jc.Status == v1.ConditionTrue {
+			return jc, nil
+		}
+	}
+
+	return commonOp.JobCondition{}, fmt.Errorf("found no current condition. Conditions: %+v", jobConditions)
+}
+
+func GetPhaseInfo(currentCondition commonOp.JobCondition, occurredAt time.Time,
+	taskPhaseInfo pluginsCore.TaskInfo) (pluginsCore.PhaseInfo, error) {
+	switch currentCondition.Type {
+	case commonOp.JobCreated:
+		return pluginsCore.PhaseInfoQueued(occurredAt, pluginsCore.DefaultPhaseVersion, "JobCreated"), nil
+	case commonOp.JobRunning:
+		return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &taskPhaseInfo), nil
+	case commonOp.JobSucceeded:
+		return pluginsCore.PhaseInfoSuccess(&taskPhaseInfo), nil
+	case commonOp.JobFailed:
+		details := fmt.Sprintf("Job failed:\n\t%v - %v", currentCondition.Reason, currentCondition.Message)
+		return pluginsCore.PhaseInfoRetryableFailure(flyteerr.DownstreamSystemError, details, &taskPhaseInfo), nil
+	case commonOp.JobRestarting:
+		details := fmt.Sprintf("Job failed:\n\t%v - %v", currentCondition.Reason, currentCondition.Message)
+		return pluginsCore.PhaseInfoRetryableFailure(flyteerr.RuntimeFailure, details, &taskPhaseInfo), nil
+	}
+
+	return pluginsCore.PhaseInfoUndefined, nil
+}
+
+func GetLogs(taskType string, name string, namespace string,
+	workersCount int32, psReplicasCount int32, chiefReplicasCount int32) ([]*core.TaskLog, error) {
+	// If kubeClient was available, it would be better to use
```

Comment:
You can use similar code to this: https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/logs/logging_utils.go#L45-L53
to avoid checking log plugins yourself... as long as you have the names of the pods...

Reply:
Done